source/connect-to-mongo/create-a-mongo-client.txt

```diff
@@ -0,0 +1,122 @@
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+To connect to a MongoDB deployment, you need two things:
+
+- A **connection URI**, also known as a *connection string*, which tells the {+driver-short+}
+  which MongoDB deployment to connect to.
+- A **MongoClient** instance, which creates the connection to the MongoDB deployment
+  and lets you perform operations on it.
+
+You can also use either of these components to customize the way the {+driver-short+} behaves
+while connected to MongoDB.
+
+This guide shows you how to create a connection string and use a ``MongoClient`` instance
+to connect to MongoDB.
+
+.. _java-rs-connection-uri:
+
+Connection URI
+--------------
+
+A standard connection string includes the following components:
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Component
+     - Description
+
+   * - ``mongodb://``
+
+     - Required. A prefix that identifies the URI as a string in the
+       standard connection format.
+
+   * - ``username:password``
+
+     - Optional. Authentication credentials. If you include these, the client
+       authenticates the user against the database specified in ``authSource``.
+
+        .. TODO: uncomment out when java-rs-auth page made
+        .. For more information about the ``authSource`` connection option, see
+        .. :ref:`java-rs-auth`. 
+
+   * - ``host[:port]``
+
+     - Required. The host and port number where MongoDB is running. If you don't
+       include a port number, the driver uses the default port, ``27017``.
+
+   * - ``/defaultauthdb``
+     
+     - Optional. The authentication database to use if the
+       connection string includes ``username:password@``
+       authentication credentials but not the ``authSource`` option. If you don't include 
+       this component, the client authenticates the user against the ``admin`` database.
+
+   * - ``?<options>``
+
+     - Optional. A query string that specifies connection-specific
+       options as ``<name>=<value>`` pairs. 
+        ..    TODO: uncomment out when connection options page made
+        ..    See :ref:`java-rs-connection-options` for a full description of
+        ..    these options.
+
+For more information about creating a connection string, see
+:manual:`Connection Strings </reference/connection-string>` in the
+MongoDB Server documentation.
+
+MongoClient
+-----------
+
+To create a connection to MongoDB, pass a connection URI as a string to the
+``MongoClient`` constructor. In the 
+following example, the driver uses a sample connection URI to connect to a MongoDB 
+instance on port ``27017`` of ``localhost``: 
+
+.. code-block:: java
+
+    import com.mongodb.reactivestreams.client.MongoClients;
+    import com.mongodb.reactivestreams.client.MongoClient;
+
+    public class MongoConnection {
+        public static void main(String[] args) {
+            String uri = "mongodb://localhost:27017/";
+            MongoClient client = MongoClients.create(uri);
```

Comment:
```suggestion
            try (MongoClient client = MongoClients.create(uri)) {
                // use `client` here
            }
```

Let's write code such that it closes the created client. Otherwise, we encourage users to not close the clients, which potentially delays the release on server resources associated with the client.

Reply:
The new code contains `client.close();` inside the `try`-with-resources statement. We should not call `client.close` ourselves, as the `try`-with-resources statement does that for us automatically. Note that the code I proposed does not have `client.close();`.

Reply:
Ah, `client.close();` was removed in a later commit.

✅ Resolved.